webapp/src/app/byz-coin.service.ts

```diff
@@ -9,6 +9,16 @@ import Dexie from "dexie";
 import { Fetcher, User } from "dynacred";
 import { Config } from "src/lib/config";
 
+function dbError(e: Error): Error {
+    return new Error("Couldn't open storage of browser.<br>This is probably due to one" +
+        " of:<br>" +
+        "- Using private browsing mode<br>" +
+        "- Setting your browser to not remember history<br>" +
+        "For more information, go to <a" +
+        " href='https://www.c4dt.org/article/failing-omniledger-login/' target='other'>OmniLedger login problems</a>" +
+        "<br><br>System error message: " + e.toString());
```

Comment:
It is not recommanded to generate HTML via typescript, as someone can insert random code here by manipulating the returned error. You should create a dedicated component which only accept an error which will be safely escaped by angular when interpolated.

Reply:
Can't I simply escape the `e.toString()` with https://stackoverflow.com/questions/24816/escaping-html-strings-with-jquery ?

Reply:
Maybe. Usually, writing yourself your own escaping library is very prone to bugs (as the mottos of "don't do your own crypto" or "don't trust your own parser"), and there is facilities already existing inside angular so why not use it direclty?

Reply:
'cause I'm lazy? Anyway, I found a third one: `don't write your own testing harness`...

Reply:
In fact the error lies even deeper: `showDialogInfo` accepts HTML code, so there must be many parts where the error might be used to inject HTML code... So I just added the `encodeURI` for the `e.toString()` here. OK?

Reply:
That won't solve the security issue, only complexify a bit the payload needed to trigger it. The secure way is to create an independent component for each message.

If you really want to keep this pattern, look at https://angular.io/guide/security#direct-use-of-the-dom-apis-and-explicit-sanitization-calls on how to use angular's calls to make it secure.

Reply:
Can you please create a commit to implement what you propose?

Reply:
Here you go, I pushed it to this branch.
I found the root issue that allowed it to work. In `lib/dialog-okcancel.html`, it uses `innerHTML` which is a huge security risk as it allow trivial XSS. As only the `dbError` calls where using HTML formating, it removed it.

Reply:
Cool - thanks a lot for taking care of this. I gotta admit I'm a bit disappointed by the slow user uptake, so I don't really care if we have a possible XSS attack from typescript-errors. Kudos to the one exploiting it ;)

But of course it's much nicer like that! Great!